### PR TITLE
Added tsv output format for bookmarks.

### PR DESCRIPTION
**Pull request by peterjschroeder**
Does this look correct to you? It is functioning.

### CODE REVIEW

**Comment by peterjschroeder** [comment]
Hold off on this. I believe it would be more useful to also allow filter by columns, but I'm trying to avoid duplicating whole functions. Also a column for Parent IDs would be useful.

File: cli/context.go [hunk 1]
```diff
@@ -253,6 +253,37 @@ func (c FFSContext) truncTableData(str string, collen int) string {
...
+func (c FFSContext) PrintPrimaryOutputTSV(data [][]string, columnFilter []int) {
+        if c.Opt.Quiet {
+                return
+        }
+
+        if len(data) == 0 {
+                return
+        }
+
+        lens := make([]int, len(data[0]))
```
Comment (Mikescher) [inline cli/context.go:265]: I think the `lens` calculation here can be removed, in the tabular output it was used for visual alignment, but that doesn't really exist in tsv

File: cli/context.go [hunk 2]
```diff
@@ -253,6 +253,37 @@ func (c FFSContext) truncTableData(str string, collen int) string {
...
+        for _, row := range data {
+                for i, cell := range row {
+                        lens[i] = mathext.Max(lens[i], len([]rune(cell)))
+                }
+        }
+
+        for rowidx := range data {
+
+                {
+                        rowstr := ""
```
Comment (Mikescher) [inline cli/context.go:275]: The only thing I'm kinda worried here is escaping the values. You are currently simply concatenating them, but imo some values can contains tabs and/or linebreaks (at least the bookmark title).

The correct way should be to use the CSVWriter from "encoding/csv" - but if you say you are not familiar enough n go we can probably just merge it like this and I'll change i later to "encoding/csv".

**Comment by Mikescher** [comment]
looks mostly good  -  I added 2 comments.  
One is about the `lens` array in your PrintPrimaryOutputTSV method (I think you can remove that part)  
The other is about using csv.Writer (but if you say you are not familar enough in go, we can merge it like this and I'll change it later).

Also would be nice if you could also update the README (there is a section that is just the --help output and teh Usage section is listing the available output formats).

Otherwise looks good, and thank 👍

**Comment by peterjschroeder** [comment]
I removed the lens code, and added to help.

Escape codes are already ignored in the values. I tested it by adding a \t to a title in my bookmarks. The output just shows \t and does not tab.

**Comment by Mikescher** [comment]
okay, looks good, I'm merging it 👍.

I think I will perhaps add tsv output to a few more commands later today and then built a new version (1.4)

**Comment by Mikescher** [comment]
@peterjschroeder Oh - one question: we are currently only printing the fields ("ID", "PARENT ID", "TYPE", "DELETED", "TITLE", "URI").

Is there any problem with printing all available fields in the tsv output (stuff like URL, Keyword, Type, etc).  
If tsv is a format that intended to get processed by another system anyway (and not directly read by a user) we can just output all potentially useful data. That way the next person already has everything

**Comment by Mikescher** [comment]
Update: okay I added the csv/tsv output to the other xxx-list subcommands. ( and also the list-tabs command from #1 )  
If you want you can try yourself at the column filtering - but I'm going home for today :)

**Comment by peterjschroeder** [comment]
Excellent. I will do the filtering tomorrow, I also ran out of time for today :P

**Comment by peterjschroeder** [comment]
Nevermind. Filtering is all you. I'm lost on how to iterate through the column filters.

**Comment by Mikescher** [comment]
Ok, I implemented a version of it with `--csv-filter`. If you pull you should get it

**Comment by peterjschroeder** [comment]
Thank you.

I was trying to approach it by just using the existing column filters, but this works too. The types in go is what throws me off. I know C, but I think python may have spoiled me away from declaring them properly. Go seems like its closer to C in the sense.

**Pull request merged**